src/util/irep_serialization.h

```diff
@@ -12,51 +12,47 @@ Date: May 2007
 #define IREP_SERIALIZATION_H_
 
 #include <map>
-#include <util/irep.h>
-
-void write_long(std::ostream &, unsigned);
-void write_string(std::ostream &, const std::string &);
+#include <util/irep2.h>
+#include "irep2_type.h"
 
+/**
+ * Class used for irep serialization, containing methods to save/load for later
+ * usage.
+ */
 class irep_serializationt
 {
 private:
-  struct ul_hash
-  {
-    unsigned short operator()(const unsigned long l) const
-    {
-      return (l & 0xFFFF);
-    }
-  };
-
-  struct ul_eq
-  {
-    bool operator()(const unsigned long l, const unsigned long r) const
-    {
-      return (l == r);
-    }
-  };
-
+  /**
+   * Helper class to generate and compare irep hashs
+   */
   struct irep_full_hash
   {
+    /**
+     * Takes an irep and generates it's hash
+     * @param i irep to be hashed
+     * @return the hash value of i
+     */
     size_t operator()(const irept &i) const
     {
       return i.full_hash();
     }
+
+    /**
+     * Compares two irep using their hashes
+     * @param i1 first irep
+     * @param i2 second irep
+     * @return truth value for if i1 hash is lower than i2 hash
+     */
     bool operator()(const irept &i1, const irept &i2) const
     {
       return i1.full_hash() < i2.full_hash();
     }
   };
 
-  struct irep_content_eq
-  {
-    bool operator()(const irept &l, const irept &r) const
-    {
-      return full_eq(l, r);
-    }
-  };
-
 public:
+  /**
+   * Helper class to keep track of which ireps where read or wrote.
```

Comment:
This sentence is unclear: "Helper class to keep track of which ireps where read or wrote"

Reply:
I am not sure what to put there.

Would "Helper class for irep serialization/unserialization to track inner ireps and strings" be better?

Reply:
It looks better.